modules/ting_fulltext/ting_fulltext.module

```diff
@@ -92,3 +102,32 @@ function ting_fulltext_ctools_plugin_directory($module, $plugin) {
     return "plugins/$plugin";
   }
 }
+
+/**
+ * Ajax call function wich returns the review in full text.
+ */
+function ting_fulltext_get_review($object_id) {
+  try {
+    $path = drupal_get_path('module', 'ting_fulltext');
+    // Include functions to parse the xml.
+    include_once $path . '/includes/ting_fulltext.pages.inc';
+
+    $xml = ting_fulltext_object_load($object_id);
+    $full_text = ting_fulltext_parse($xml);
+    // We dont need to display the title again.
+    unset($full_text['title']);
+
+    $results = array(
+      '#theme' => 'ting_fulltext',
+      '#fields' => $full_text,
+    );
+    $output = drupal_render($results);
+
+    return drupal_json_output($output);
+  }
+  catch (Exception $e) {
+    watchdog_exception('ting_fulltext', $e);
+    drupal_add_http_header('Status', '500 Internal Server Error');
+    drupal_exit();
+  }
```

Comment:
I am not sure you need to do this error handling yourself. If an exceptions bubbles up from a page callback Drupal should log the error and return a error code 500 response anyway.